.pre-commit-config.yaml

```diff
@@ -0,0 +1,44 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v3.2.0
+    hooks:
+    -   id: trailing-whitespace
+    -   id: end-of-file-fixer
+    -   id: check-yaml
+    -   id: check-added-large-files
+
+    # Clang-format for C++
+    # This brings in a portable version of clang-format.
+    # See also: https://github.com/ssciwr/clang-format-wheel
+  - repo: https://github.com/pre-commit/mirrors-clang-format
+    rev: v18.1.8
+    hooks:
+    - id: clang-format
+      types_or: [c++, c]
+
+    # Clang-tidy for C++
+    # WARN: repo may not maintained anymore since 2021.
+  - repo: https://github.com/pocc/pre-commit-hooks
+    rev: v1.3.5
+    hooks:
+      - id: clang-tidy
+        # Currently disabled, see:
+        # https://github.com/beman-project/exemplar/pull/34#discussion_r1784958937
+        stages: [manual]
+        additional_dependencies: ['clang-tidy==18.1.8']
+
```

Comment:
```suggestion
```

I think we can remove this given the problems you've been seeing.

Reply:
Thanks for the review.

The main reason for choosing `gersemi` is because it seems to be the only CMake formatter I found that is being actively maintained. We will need to keep an eye on these tools as they still haven't released a major version yet.

Thanks for the link to the carbon pre-commit instructions, I don't have experience writing good documentations so this is very helpful.
The reason I have not written the documentation is I am waiting for more review on the tool chain, I can write the documentation after more maintainer agrees on the contents of this pr. I would be open to putting the documentation in a separate pr.